metricbeat/module/kubernetes/state_pod/state_pod.go

```diff
@@ -94,17 +89,28 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	}, nil
 }
 
-// Fetch methods implements the data gathering and data conversion to the right format
-// It returns the event which is then forward to the output. In case of an error, a
-// descriptive error must be returned.
-func (m *MetricSet) Fetch() ([]common.MapStr, error) {
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
 	m.enricher.Start()
 
 	events, err := m.prometheus.GetProcessedMetrics(mapping)
-	if err == nil {
-		m.enricher.Enrich(events)
+	if err != nil {
+		m.Logger().Error(err)
+		reporter.Error(err)
+		return
 	}
-	return events, err
+
+	m.enricher.Enrich(events)
+	for _, event := range events {
+		reporter.Event(mb.Event{
+			MetricSetFields: event,
```

Comment:
You should use `RootFields` for common fields as `kubernetes.namespace` or `kubernetes.pod.name`.